Formula/s2geometry.rb

```diff
@@ -19,69 +18,108 @@ class S2geometry < Formula
   end
 
   depends_on "cmake" => :build
-  depends_on "glog" => :build
-  depends_on "googletest" => :build
+  depends_on "abseil"
+  depends_on "glog"
   depends_on "openssl@1.1"
 
+  on_linux do
+    depends_on "gcc"
+  end
+
+  fails_with gcc: "5" # C++17
+
   def install
+    # Abseil is built with C++17 and s2geometry needs to use the same C++ standard.
+    inreplace "CMakeLists.txt", "set(CMAKE_CXX_STANDARD 11)", "set(CMAKE_CXX_STANDARD 17)"
```

Comment:
Docs mention using `-DCMAKE_CXX_STANDARD=17`, but it doesn't work due to how value is set. I guess another option would be changing upstream CMakeLists.txt to use a `CACHE` value.